### PR TITLE
Improve categorizing for download stats and add unit tests

### PR DESCRIPTION
**Pull request by aiuto**
Fix bug in categorizing rules_docker-v0.9.0.tar.gz ￼…
Handle files of the form bazel-toolchains-e6b0efb.tar.gz

### CODE REVIEW

File: categorize.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+import collections
+import re
+import string
+import sys
+
+# The bins we categorize artifacts as
+Bins = collections.namedtuple(
```
Comment (jin) [inline categorize.py]: can we rename to "executable" or "binaries"? I thought this meant bin as in bin-packing, on the first read.

**Comment by aiuto** [comment]
I do mean bins, as in categories.  Buckets might also work. I can change to
that.

File: categorize.py (hunk 1 above)
Comment (aiuto) [reply categorize.py]: It actually does mean bins as in categories, not binaries. I changed it to 'buckets' which is less ambiguous when we are talking about categorizing into buckets.

**Review by jin — APPROVED** [review]
thanks!

**Pull request merged**